### ISSUE TITLE
Spreadsheet: No way to reset cell background color

### ISSUE THREAD

**Issue by NXTler** [FreeCAD/FreeCAD#13862]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Problem description

Greetings,
when changing the background color of a cell, there is no way to revert to transparent again without deleting the cell:
![grafik](https://github.com/FreeCAD/FreeCAD/assets/73062438/eec405c0-2382-4696-a74c-c4e02ab46fb8)
FreeCAD also sets the background to white when using STRG + Z, assuming that the spreadsheet background is the same, which is not the case when the user uses a dark theme.

### Full version info

```shell
OS: Manjaro Linux (KDE/plasma)
Word size of FreeCAD: 64-bit
Version: 0.22.0dev.37213 (Git) AppImage
Build type: Release
Branch: main
Hash: 20e7deb86a8c6c2cd2378f09f8313760933f3a5c
Python 3.11.9, Qt 5.15.13, Coin 4.0.2, Vtk 9.2.6, OCC 7.7.2
Locale: German/Germany (de_DE)
Installed mods: 
  * PieMenu 1.2.7
  * sheetmetal 0.4.2
  * A2plus 0.4.63
  * QuickMeasure 2022.10.28
  * Curves 0.6.23
  * CfdOF 1.25.2
  * IconThemes
  * fasteners 0.5.12
```

### Subproject(s) affected?

Spreadsheet

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by hyarion** [comment]
I've started to work on this issue.

What I've notice is that there is partial support for clearing colors already, but there are some quirks that makes clearing background impossible.

* Cells that has the color `rgbaF(1, 1, 1, 0)` will clear the `BACKGROUND_COLOR_SET` flag: https://github.com/FreeCAD/FreeCAD/blob/main/src/Mod/Spreadsheet/App/Cell.cpp#L506
* ColorPicker can create custom colors with alpha: https://github.com/FreeCAD/FreeCAD/blob/main/src/Mod/Spreadsheet/Gui/qtcolorpicker.cpp#L914
* When the workbench gets the color from the color picker, it will discard the alpha value: https://github.com/FreeCAD/FreeCAD/blob/main/src/Mod/Spreadsheet/Gui/Workbench.cpp#L180

There are multiple ways to fix this.
1. We could change the clear-color to opaque white - `rgbaF(1, 1, 1, 1)`. But then you wouldn't be able to use white as background in dark mode. It is a quick fix and works as expected in light mode. It would be a bit strange in dark mode, but users should be able to find it.
2. We could also add a new standard color in the background color picker, which is this transparent white `rgbaF(1, 1, 1, 0)` which could be called `"None"`. It would be a bit strange to have a transparent color in the color picker so I would suggest that we indicate that this is a remove-background-color. This would also make it possible to create semi-transparent colors and if we support that, then we need to create a way to show those colors clearly too.

Option 1 is easy to fix with minimal change and I think it could be enough for v1.0. (Implemented in this branch: https://github.com/hyarion/FreeCAD/tree/spreadsheet-use-opaque-white-clear-color)
Option 2 requires some UI work but I think this is probably the way to go in the long run (I've implemented it with programmers art for the ui in this branch: https://github.com/hyarion/FreeCAD/tree/spreadsheet-clear-bg-and-transparent-colors)

Here's a screen recording of the Option 2 branch:

https://github.com/user-attachments/assets/97297e6e-21eb-46ee-b5c7-9887bdb030ae

And a screenshot of the programmers-art-color-swatch with clear color and semi-transparency:

<img width="133" alt="image" src="https://github.com/user-attachments/assets/cd8dec6f-d5c7-4677-b3db-6bbc2acfdd98">


So what do you think, should we:
1. Remove the possibility to change background color in cells to white, but allowing users to clear the color and maybe get a fix ready for v1.0? Or,
2. Keep white as a background color option and add new none-color with a special icon?
3. Something else completely?

**Comment by NXTler** [comment]
I would go with a none colour option. I can imagine that some people might use it even in none light mode themes, so keeping that functionality is important.

**Comment by hyarion** [comment]
@NXTler I agree, it feels a bit like breaking something else by fixing this.

To continue with the "none"-color option, we need to decide on:
* Do we want to support semi-transparent colors in spreadsheets?
* Is the look I created good enough, or should we redesign the color picker?
* If redesign, how should it look?

@luzpaz How do I get attention on this from the right people? Or should I lay low until 1.0 is released?

**Comment by NXTler** [comment]
This issue is just for the ability to set a none colour. However, if you would like to redesign the colour selector anyway here are my thoughts:
I think the best approach is to do something similar to Office programs and offer all the basic colours in different strengths on a grid. It's intuitive and people know it well:
<img width="173" alt="82b36c59-818d-41cb-99c4-079df967f487-758406764" src="https://github.com/user-attachments/assets/d8b7660e-975d-41b7-a3c2-4d032fdb9bc4">

But different from these office programs, the strengths should be set through their transparency value rather then their white content. That way the new selector should smoothly integrate into all themes without causing any unexpected discolouration or dark/light theme overlaps.

**Comment by NXTler** [comment]
That being said, the design you created is more than good enough and would already fix this issue.

**Comment by MisterMakerNL** [comment]
If you got the time can you check [this](https://github.com/FreeCAD/FreeCAD/issues/8168) out too?

**Comment by luzpaz** [comment]
Is this still an issue ?

**Comment by hyarion** [comment]
Yes it is

**Comment by NXTler** [comment]
1.0 is out now, so I think it could be worked on again, if anyone wants to.
